web/react/components/modal.jsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+export default class Modal extends ReactBootstrap.Modal {
```

Comment:
This seems unnecessarily complicated. The onPreshow method is never used. And what's the point because the parent has already decided to show the modal and is past it's componentWillMount function. 
The onShow method is just checking if the state has changed from show from not being shown. This is something the parent component can easily do.